plugins/by-name/scrollview/default.nix

```diff
@@ -0,0 +1,450 @@
+
+      cursor_priority = mkPriorityOption "cursor signs" 0;
+
+      cursor_symbol = mkNullOrStr ''
+        The symbol for cursor signs.
+
+        Defaults to a small square, resembling a block cursor.
+      '';
+
+      diagnostics_error_priority = mkPriorityOption "cursor diagnostic error signs" 60;
+
+      diagnostics_error_symbol = mkNullOrOption symbolType ''
+        The symbol for diagnostic error signs.
+
+        Defaults to the trimmed sign text for `DiagnosticSignError` if defined, or `"E"` otherwise.
+
+        ${symbolCommonDesc}
+      '';
+
+      diagnostics_hint_priority = mkPriorityOption "the diagnostic hint signs" 30;
+
+      diagnostics_hint_symbol = mkNullOrOption symbolType ''
+        The symbol for diagnostic hint signs.
+
+        Defaults to the trimmed sign text for `DiagnosticSignHint` if defined, or `"H"` otherwise.
+
+        ${symbolCommonDesc}
+      '';
+
+      diagnostics_info_priority = mkPriorityOption "the diagnostic info signs" 40;
+
+      diagnostics_info_symbol = mkNullOrOption symbolType ''
+        The symbol for diagnostic info signs.
+
+        Defaults to the trimmed sign text for `DiagnosticSignInfo` if defined, or `"I"` otherwise.
+
+        ${symbolCommonDesc}
+      '';
+
+      diagnostics_severities =
+        mkNullOrOption
+          (
+            with types;
+            oneOf [
+              ints.unsigned
+              rawLua
+              (listOf (either ints.unsigned rawLua))
+            ]
+          )
+          ''
+            List of numbers specifying the diagnostic severities for which signs will be shown.
+            The default includes `vim.diagnostic.severity.ERROR`, `vim.diagnostic.severity.HINT`,
+            `"vim.diagnostic.severity.INFO"`, and `vim.diagnostic.severity.WARN`.
+
+            Considered only when the plugin is loaded.
+          '';
+
+      diagnostics_warn_priority = mkPriorityOption "the diagnostic warn signs" 50;
+
+      diagnostics_warn_symbol = mkNullOrOption symbolType ''
+        The symbol for diagnostic warn signs.
+
+        Defaults to the trimmed sign text for `DiagnosticSignWarn` if defined, or `"W"` otherwise.
+
+        ${symbolCommonDesc}
+      '';
+
+      folds_priority = mkPriorityOption "fold signs" 30;
+
+      folds_symbol = mkNullOrOption symbolType ''
+        The symbol for fold signs.
+
+        Defaults to a right-pointing triangle.
+
+        ${symbolCommonDesc}
+      '';
+
+      latestchange_priority = mkPriorityOption "latestchange signs" 10;
+
+      latestchange_symbol = mkNullOrStr ''
+        The symbol for latestchange signs.
+
+        Defaults to a Greek uppercase delta.
+      '';
+
+      loclist_priority = mkPriorityOption "loclist signs" 45;
+
+      loclist_symbol = mkNullOrOption symbolType ''
+        The symbol for loclist signs.
+
+        Defaults to a small circle.
+
+        ${symbolCommonDesc}
+      '';
+
+      marks_characters = mkNullOrOption (with types; either rawLua (listOf str)) ''
```

Comment:
```suggestion
      marks_characters = mkListOf types.str ''
```

Reply:
I think this one is still relevant, `maybeRaw (listOf (maybeRaw str)` is produced by `defaultNullOpts.mkListOf str`.

If you don't want a pluginDefault you could use `mkListOf'`

```nix
defaultNullOpts.mkListOf' {
  type = types.str;
  description = "";
}
```

If you really don't want that, you can add maybeRaw to the element type yourself: `maybeRaw (listOf (maybeRaw str)`.

Reply:
I went with the latter as for me the goal of `defaultNullOpts` helper is to provide a plugin default.
Using them and deleting their purpose (the plugin default) doesn't seem right to me.

To be honest, your solution is elegant too. I don't feel too hard about it.

Reply:
> the goal of `defaultNullOpts` helper is to provide a plugin default.

I disagree with that sentiment, but I don't want to get into a debate here :grin: 

(I think that was the original purpose, but they have evolved over time to relate to anything that is maybeRaw and defaults to null)

Reply:
Actually, I'm curious, I changed it to `mkListOf` but I can't manage to get the _Plugin default: `null`_ out of the documentation.

Reply:
```nix
      marks_characters = defaultNullOpts.mkListOf' {
        type = types.str;
        # default = null;
        description = ''
          List of strings specifying characters for which mark signs will be shown.
          Defaults to characters `a-z` and `A-Z`.

          Considered only when the plugin is loaded.
        '';
      };
```

Reply:
> but I can't manage to get the _Plugin default: `null`_ out of the documentation.

Interesting... That sounds like a bug I should investigate :thinking:

Reply:
I guess I will merge as is then.

Reply:
My apologies. I think that I got tricked by `direnv`. The `mkListOf'` implementation does work fine (and does not show _plugin default: `null`_):
![image](https://github.com/user-attachments/assets/5f7fb6f8-c2b3-4116-a72d-e685e863be37)

Sorry for the false alarm.